test/test_python_dispatch.py

```diff
@@ -55,6 +55,23 @@ def wrap(e):
         logging.getLogger("LoggingTensor").info(f"{func.__module__}.{func.__name__}", args, rs)
         return rs
 
+class FiniteTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        assert torch.isfinite(elem).all()
+        return torch.Tensor._make_subclass(cls, elem, elem.requires_grad)
+
+    def __repr__(self):
+        return f"FiniteTensor({super().__repr__})"
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        # TODO: also explicitly recheck invariants on inplace/out mutation
```

Comment:
Good point - at least I don't think applies to too many ops, since most ops that mutate their inputs also return them as outputs

Reply:
There's not a good idiom for querying this information from torch dispatch, unlike in boxed fallback, so we kind of need some of the same knobs to schema that you get there.